packages/server/src/file/file.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@nestjs/common';
+import { InjectModel } from '@nestjs/mongoose';
+import { File, FileDocument } from './file.model';
+import { Model } from 'mongoose';
+import { CreateFileInput } from './file.dto';
+import { Comment, CommentDocument } from 'src/comment/comment.model';
+
+@Injectable()
+export class FileService {
+  constructor(
+    @InjectModel(File.name) private fileModel: Model<FileDocument>,
+    @InjectModel(Comment.name) private commentModel: Model<CommentDocument>
+  ) {}
+
+  async create(createFileInput: CreateFileInput): Promise<File> {
+    const createFile = new this.fileModel(createFileInput);
+    return createFile.save();
+  }
+
+  findByFileId(id: string): Promise<File | null> {
+    return this.fileModel
+      .findOne({ fileId: id })
+      .populate({ path: 'comments', populate: { path: 'replies' } })
```

Comment:
The challenge with populating here is that the types clash since its no longer a list of object ids